lib/update/cluster/builder.go

```diff
@@ -271,6 +273,68 @@ func (r phaseBuilder) openEBS(leadMaster storage.UpdateServer) *update.Phase {
 	return &phase
 }
 
+// openEBSDataPlane checks if existing OpenEBS pools or volumes need to be upgraded
+// and generates upgrade steps
+func (r phaseBuilder) openEBSDataPlane(ctx context.Context, storageAppVersion string, root *update.Phase) error {
+	pv, err := kubectl.OpenEBSPoolsVersions(ctx)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	openEBSPhase("pool", pv, updateOpenEBSPool, storageAppVersion, root)
+
+	vv, err := kubectl.OpenEBSVolumesVersions(ctx)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	openEBSPhase("volume", vv, updateOpenEBSVolume, storageAppVersion, root)
+
+	return nil
+}
+
+func openEBSPhase(phase string, components map[string]string, executor string, storageAppVer string, root *update.Phase) {
+	for k, v := range components {
+		toVer := openEBSComponentToVer(storageAppVer, k, v)
+		if toVer == "" {
+			continue
+		}
+
+		phase := update.Phase{
+			ID:          fmt.Sprintf("openebs-%v-%v", phase, k),
+			Description: fmt.Sprintf("Update OpenEBS %v: %v to %v", phase, k, toVer),
+			Executor:    executor,
+			Data:        &storage.OperationPhaseData{Data: buildOpenEBSUpgradePhaseData(k, v, toVer)},
+		}
+		root.AddSequential(phase)
+	}
+}
+
+func buildOpenEBSUpgradePhaseData(dataPlaneComponent string, fromVer string, toVer string) string {
+	return dataPlaneComponent + " " + fromVer + " " + toVer
+}
+
+func openEBSComponentToVer(storageAppVer string, openEBSComponentName string, openEBSComponentFromVer string) string {
+	if storageAppVer == "0.0.4" {
+		openEBSComponentToVer := "2.2.0"
```

Comment:
This value seems a bit buried, what if an upstream patch comes out and we need to bump this to 2.2.1. Ideally we would have a Makefile variable so it's easy to locate at bump a patch, and as long as the upgrade behaviour doesn't change, we can just update the variable / storage-app version and release.